pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
we already have an internal metrics lib, so using prometheus lib makes little sense.

Think just adding a new metrics api resource that returns data in prometheus format should be a good solution right? we have a metrics endpoint at /api/metrics , maybe add another one at /api/metrics/prometheus that supports the prometheus format

> Prometheus can't really work with a data model like this. We need real counters.

I hate counters that never reset, terrible for anything other than rates. And rates are terrible for business/user/behavior metrics where you want accurate sums over different time periods. If prometheus can't work with count's then I guess only the timers that Grafana sends are of interest.